source/slang/parser.cpp

```diff
@@ -2199,14 +2199,44 @@ namespace Slang
                 paramConstraint->sup = parser->ParseTypeExp();
 
                 AddMember(assocTypeDecl, paramConstraint);
-                if (!AdvanceIf(parser, TokenType::Comma))
-                    break;
-            }
+            } while (AdvanceIf(parser, TokenType::Comma));
         }
         parser->ReadToken(TokenType::Semicolon);
         return assocTypeDecl;
     }
 
+    RefPtr<RefObject> ParseGlobalGenericParamDecl(Parser * parser, void *)
+    {
+        RefPtr<GlobalGenericParamDecl> genParamDecl = new GlobalGenericParamDecl();
+        auto nameToken = parser->ReadToken(TokenType::Identifier);
+        genParamDecl->nameAndLoc = NameLoc(nameToken);
+        genParamDecl->loc = nameToken.loc;
+        if (AdvanceIf(parser, TokenType::Colon))
```

Comment:
We should probably consider hoisting out a distinct subroutine for parsing trailing generic type constraints, that we can share between global generics and assocaited types.

Also: I'm trying to shift the naming convention to have `lowerCamelCase` for functions (and ideally all *values*, although I'm waffling when it comes to `enum` tags), so can this be `parseGlobalGenericParamDecl`?